### DISCUSSION TITLE
A small number of videos with a pending status appear, untranscoded and uncompressed.

### DISCUSSION THREAD

**Discussion by tippytea** [mediacms-io/mediacms#568] [Q&A]
A small number of videos with a pending status appear, untranscoded and uncompressed. However, most of the time mediacms can finish pressing and transcoding, and only a few cannot transcode and have a pending status. Currently I have over 1000 videos uploaded to my mediacms, and about 70 of them have a pending status.

My question is: Is there a quick way to submit these videos that have been uploaded but have become pending again for transcoding? For example, enter a command to have all the videos with a pending status transcoded again?

Thanks to the mediacms creative team, great program.

**Comment by mgogoulos** [comment]
Hi, 

there's no way to do this from the web interface now. That said, this is something that could be part of the 'manage media' page for admins, and even provide logs on the media page why these transcodings have failed.
 
It could also be a Django management command.

For now, if you can enter the Django shell, you can ru the following to re-start the transcoding procedure to videos that didn't finish well:

```
from files.models import Media                                                                                                                                                                                            
for media in Media.objects.filter(media_type='video').exclude(encoding_status='success'): 
    media.media_init() 
```

Eg 

```
root@markos /home/mediacms.io/mediacms # source ../bin/activate
(mediacms.io) root@markos /home/mediacms.io/mediacms # python manage.py shell
Python 3.6.8 (default, Oct  7 2019, 12:59:55) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from files.models import Media                                                                                                                                                                                            
In [15]: for media in Media.objects.filter(media_type='video').exclude(encoding_status='success'): 
    ...:     media.media_init() 

```

**Comment by mgogoulos** [comment]
Chances are that some of them could eventually be transcoded correctly but some/most will fail for the same reason the failed initially, this is why logs have to be provided on an easy way on the interface (you can see on logs/ what's wrong but this is not intuitive).

Interesting to hear about your portal, care to share URL or mention what is the sector/area? 

Cheers
